vortex-serde/src/layouts/read/filtering.rs

```diff
@@ -33,6 +35,19 @@ impl RowFilter {
         for expr in self.conjunction.iter() {
             let new_mask = expr.evaluate(target)?;
             mask = and(new_mask, mask)?;
+
+            if mask
+                .clone()
+                .into_bool()?
+                .statistics()
+                .compute_true_count()
+                .unwrap_or_default()
+                == 0
```

Comment:
Do you want to compute the true_count here? Anyway to access stats you don't need to canonicalize the arrays

Reply:
Computing makes sense IMO, canonicalization is a leftover from using the `BooleanBuffer::count_set_bits`, will remove it.